### ISSUE TITLE
[Feature Request] Add xem names to "expected_titles" option in guessit

### ISSUE THREAD

**Issue by darthShadow** [pymedusa/Medusa#3927]
Example where this is necessary:

**Without:**
```
➜ guessit "[Chyuu] Nanatsu no Taizai - Imashime no Fukkatsu - 10 [720p][8BC6417A].mkv"

For: [Chyuu] Nanatsu no Taizai - Imashime no Fukkatsu - 10 [720p][8BC6417A].mkv
GuessIt found: {
    "release_group": "Chyuu",
    "title": "Nanatsu no Taizai",
    "episode_title": "Imashime no Fukkatsu",
    "episode": 10,
    "screen_size": "720p",
    "crc32": "8BC6417A",
    "container": "mkv",
    "mimetype": "video/x-matroska",
    "type": "episode"
}
```

**With:**
```
➜ guessit "[Chyuu] Nanatsu no Taizai - Imashime no Fukkatsu - 10 [720p][8BC6417A].mkv" -T "Nanatsu no Taizai" -T "Nanatsu no Taizai - Imashime no Fukkatsu"

For: [Chyuu] Nanatsu no Taizai - Imashime no Fukkatsu - 10 [720p][8BC6417A].mkv
GuessIt found: {
    "release_group": "Chyuu",
    "title": "Nanatsu no Taizai - Imashime no Fukkatsu",
    "episode": 10,
    "screen_size": "720p",
    "crc32": "8BC6417A",
    "container": "mkv",
    "mimetype": "video/x-matroska",
    "type": "episode"
}
```

This may seem related to the line related issue and I have already opened an [issue](https://github.com/guessit-io/guessit/issues/537) in the guessit repo. But, I think this functionality should be included here too.

**Comment by p0psicles** [comment]
There is already an automated system for this in medusa.
We run some regex on series titles and exceptions and decide based on that if it's a "tricky" title that should be added to the expected_titles. We could add a regex check for something like: `title - some more title`

**Comment by darthShadow** [comment]
Yeah, saw that in the `guessit_parser.py` file. However, and correct me if I am wrong, it doesn't seem to include the xem mappings, only the main name and the exceptions.

This should also solve the multitude of line issues related to guessit that have been mentioned in the other threads. And I have tried the latest dev version of guessit too, this doesn't work even with the fix for the line issue.

**Comment by p0psicles** [comment]
Well we have to be careful there. As allot of the `line` issues have already been fixed in guessit v3. We're currently still on v2. We're waiting for v3 to have been released. And the guessit devs haven't responded yet.

So more often the expected titles `feature` is more like a workaround.

**Comment by darthShadow** [comment]
As per the latest guessit version, shows such as `[HorribleSubs] Yowamushi Pedal - Glory Line - 01 [1080p].mkv` will be parsed as follows:
```
➜ guessit "[HorribleSubs] Yowamushi Pedal - Glory Line - 01 [1080p].mkv"

For: [HorribleSubs] Yowamushi Pedal - Glory Line - 01 [1080p].mkv
GuessIt found: {
    "release_group": "HorribleSubs",
    "title": "Yowamushi Pedal",
    "alternative_title": "Glory Line",
    "episode": 1,
    "screen_size": "1080p",
    "container": "mkv",
    "mimetype": "video/x-matroska",
    "type": "episode"
}
```

Even this is not correct since it's being returned as an `alternative_title` instead of as a part of the series `title` itself. Is this the expected behaviour and some extra handling will be added for the `alternative_title` field in medusa?

My thinking process was that medusa should not be wholly dependent on guessit for the parsing. If in the future we have an anime that is split via something else instead of `-`, then we will have to again go back to guessit for the modifications.

**Comment by p0psicles** [comment]
I was in the impression that the title would be parsed as one string now. Maybe @ratoaq2 can comment on it?

**Comment by ratoaq2** [comment]
I believe in medusa it's in one string for animes. Guessit is and will continue to be in multiple strings.
The dash separator will flag that you have a title plus an alternate title. You can find all the scenarios in the yaml test file in medusa and also in guessit

**Comment by ratoaq2** [comment]
Some examples

https://github.com/pymedusa/Medusa/blob/develop/tests/test_guessit.yml#L2362

https://github.com/pymedusa/Medusa/blob/develop/tests/test_guessit.yml#L2428

**Comment by darthShadow** [comment]
@p0psicles, I was thinking of giving this a try for a PR. I think the following change should be sufficient in [name_parser/guessit_parser.py](https://github.com/pymedusa/Medusa/blob/d6eb72d4e1019e5d1286ae6ac1d7e4b318229ebe/medusa/name_parser/guessit_parser.py#L108-L109):

```py
if not any([char.isdigit() for char in series]):
	continue
```

replaced with

```py
if all([char.isalpha() for char in series]):
	continue
```

Let me know if something else also needs to be changed.

**Comment by OmgImAlexis** [comment]
@darthShadow either way you we should wait for guessit v3 before more changes are done to the parsing.

Ideally Medusa shouldn't need to know anything about parsing as Guessit should handle it all.
